pkg/provider/azure_loadbalancer.go

```diff
@@ -890,17 +861,18 @@ func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service,
 	if name, found := service.Annotations[consts.ServiceAnnotationPIPName]; found && name != "" {
 		return name, true, nil
 	}
+	isIPv6 := utilnet.IsIPv6String(service.Spec.ClusterIP)
 	if ipPrefix, ok := service.Annotations[consts.ServiceAnnotationPIPPrefixID]; ok && ipPrefix != "" {
-		return az.getPublicIPName(clusterName, service), false, nil
+		return az.getPublicIPName(clusterName, service, isIPv6), false, nil
```

Comment:
since `service` is part of parameters already, would it make sense moving L864 inside `getPublicIPName()` and hence no need another parameter?

Reply:
This is not the final code of dualstack support. Please take a look at the code including everything. `isIPv6` will be a parameter of `determinePublicIPName()` and it only handles one IPFamily in one time.
https://github.com/lzhecheng/cloud-provider-azure/blob/9dc86bcf502be2cf384160be130ae5e5b53aa498/pkg/provider/azure_loadbalancer.go#L897-L904

Reply:
This code design makes less code change. Take `determinePublicIPName()` as an example. `reconcilePublicIP()` calls `determinePublicIPName()`, `getPublicIPUpdates()`, `ensurePublicIPExists()`. With current code design, only lines calling these methods are changed (add a parameter isIPv6). If these methods handle two IP families, they will return 2 desiredPipName, discoveredDesiredPublicIP, deletedDesiredPublicIP, etc. Extra logic is needed to handle something like if returned values are empty or not OR the order of 2 returned values. 
If some calls involve calling API, to reduce API call, I will handle 2 IP families together.